Match.cpp

```diff
@@ -331,21 +331,115 @@ void Match::print( ostream& fout )
-		fout << endl;
+		if ( ALL_SAMPLES.useEmbarkRFGermlineOutput ) {
+			int key1 = stoi(node[0]->single_id);
+			int key2 = stoi(node[1]->single_id);
+			vector<string> oline;
+			string joined_oline;
+			ofstream* ofs;
+			string chromosome = ALL_SNPS.getSNP(snp_start).getChr();
+			string start_pos = to_string(ALL_SNPS.getSNP(snp_start).getPhysPos());
+			string end_pos = to_string(ALL_SNPS.getSNP(snp_end).getPhysPos());
+			
+			// homoz
+			if ( key1 == key2 && node[0]->is_new ) {
+				//cout << "Writing out homoz" << endl;
+				oline.push_back(node[0]->single_id);
+				oline.push_back(node[0]->haplotype);
+				oline.push_back(node[1]->single_id);
+				oline.push_back(node[1]->haplotype);
+				oline.push_back(chromosome);
+				oline.push_back(start_pos);
+				oline.push_back(end_pos);
+				ofs = node[0]->getIndividualHomozFile();
+				join(oline, '\t', joined_oline);
+				*ofs << joined_oline << endl;
+			}
+			// key1 is new, key2 is old
+			else if ( key1 > key2 && node[0]->is_new ) {
+				//cout << "Writing match tracts for new key1 " << key1 << endl;
+				oline.push_back(node[0]->single_id);
+				oline.push_back(node[0]->haplotype);
+				oline.push_back(node[1]->single_id);
+				oline.push_back(node[1]->haplotype);
+				oline.push_back(chromosome);
+				oline.push_back(start_pos);
+				oline.push_back(end_pos);
+				ofs = node[0]->getIndividualMatchFile();
+				join(oline, '\t', joined_oline);
+				*ofs << joined_oline << endl;
+			}
+			// key2 is new, key1 is old
+			else if ( key2 > key1 && node[1]->is_new ) {
+				//cout << "Writing match tracts for new key2 " << key2 << endl;
+				oline.push_back(node[1]->single_id);
+				oline.push_back(node[1]->haplotype);
+				oline.push_back(node[0]->single_id);
+				oline.push_back(node[0]->haplotype);
+				oline.push_back(chromosome);
+				oline.push_back(start_pos);
+				oline.push_back(end_pos);
+				ofs = node[1]->getIndividualMatchFile();
+				join(oline, '\t', joined_oline);
+				*ofs << joined_oline << endl;
+			}
+			// newdog : newdog comparison, write out same record twice
+			else if ( node[0]->is_new  &&  node[1]->is_new  && key1 != key2 ) {
+				//cout << "Writing records for both keys";
+				oline.push_back(node[0]->single_id);
+				oline.push_back(node[0]->haplotype);
+				oline.push_back(node[1]->single_id);
+				oline.push_back(node[1]->haplotype);
+				oline.push_back(chromosome);
+				oline.push_back(start_pos);
+				oline.push_back(end_pos);
+				ofs = node[0]->getIndividualMatchFile();
+				join(oline, '\t', joined_oline);
+				*ofs << joined_oline << endl;
+
+				oline.push_back(node[1]->single_id);
+				oline.push_back(node[1]->haplotype);
+				oline.push_back(node[0]->single_id);
+				oline.push_back(node[0]->haplotype);
+				oline.push_back(chromosome);
+				oline.push_back(start_pos);
+				oline.push_back(end_pos);
+				ofs = node[1]->getIndividualMatchFile();
+				join(oline, '\t', joined_oline);
+				*ofs << joined_oline << endl;
+			}
```

Comment:
else raise exception? Something is serious wrong if none of the above are true

Reply:
done